### PR TITLE
Improve extension config

### PR DESCRIPTION
**Pull request by ehuelsmann**
@pjlsergeant this commit makes the configuration of extensions in the config file into a hash of hashes.

The code didn't get much clearer, unfortunately. Otoh, I hope that specializing the 'extensions' configuration key helps keep things clean 'enough'.

After this one, there's another PR waiting which fixes an issue I found by coding my extension(s). Please hold releasing 0.48 until we have this *and* the other one discussed. If you want to merge it yourself: the change is sitting on my 'load-extension-steps' branch.

### CODE REVIEW

File: lib/App/pherkin.pm [hunk 1]
```diff
@@ -192,10 +192,17 @@ sub _load_config {
...
+            if ( $key ne 'extensions' ) {
+                die $profile_problem->(
 "Option $key is a [$reftype] but can only be a single value or ARRAY"
-            ) unless $reftype eq 'ARRAY';
-            push( @arguments, $key, $_ ) for @$value;
+                ) unless $reftype eq 'ARRAY';
+                push( @arguments, $key, $_ ) for @$value;
+            } else {
+                die $profile_problem->(
+                    "Option $key is a [$reftype] but can only be a HASH" )
```
Comment (pjlsergeant) [inline lib/App/pherkin.pm]: Probably worth adding a slightly more expansive message there ... 

```perl
. "as 'extensions' is a special case - see the documentation for details"
```

Which reminds me, this probably needs documenting too :-)

Comment (ehuelsmann) [inline lib/App/pherkin.pm]: Agreed as to adding docs. Where is the config documentation at the moment? (I looked for it, but couldn't find it.)

Commit for the enhanced message coming up.

**Comment by pjlsergeant** [comment]
`bin/pherkin` - "=head1 CONFIGURATION PROFILES"

**Comment by pjlsergeant** [comment]
OK, merged. Do you want me to mint a new release?

**Comment by ehuelsmann** [comment]
Just sent a new pull request for loading the steps indicated by extensions. Please evaluate that for inclusion in the coming release too.

**Pull request merged**